pkg/autodiscovery/providers/docker.go

```diff
@@ -120,16 +120,18 @@ func (d *DockerConfigProvider) IsUpToDate() (bool, error) {
 func parseDockerLabels(containers map[string]map[string]string) ([]integration.Config, error) {
 	var configs []integration.Config
 	for cID, labels := range containers {
-		c, err := extractTemplatesFromMap(docker.ContainerIDToEntityName(cID), labels, dockerADLabelPrefix)
-		switch {
-		case err != nil:
-			log.Errorf("Can't parse template for container %s: %s", cID, err)
-			continue
-		case len(c) == 0:
+		c, errors := extractTemplatesFromMap(docker.ContainerIDToEntityName(cID), labels, dockerADLabelPrefix)
+
+		if len(c) == 0 {
```

Comment:
If `extractTemplatesFromMap` returns no config but errors, will these be logged if we continue here?

Reply:
Yes thanks, that's a bug. 
That's not necessary because to check length and `continue` since append a nil no-op